src/iterators.js

```diff
@@ -4,3 +4,9 @@ export let differentLast = (normalCase, lastCase) => (acc, i, list) =>
   i === list.length - 1
     ? _.iteratee(lastCase)(acc, i, list)
     : _.iteratee(normalCase)(acc, i, list)
+
+export let differentIndex = (normalCase, indexCase, index) =>
+(acc, i) =>
+  i === index
+    ? _.iteratee(indexCase)(acc, i)
```

Comment:
why would `differentIndex` not pass `list` in as the third parameter like `differentLast`?

Reply:
probably because it's not being used to calculate whether we're on the last index - but yes, the iterator should pass all of its arguments through to the case functions. 

I'd actually suggest adding a spread for any extra args to both `differentIndex` and `differentLast`, primarily for correctness. We can also take the opportunity to play a little golf with the ternary:

```js
export let differentIndex = (normalCase, indexCase, index) => (acc, i, ...args) =>
  _.iteratee(i === index ? indexCase : normalCase)(acc, i, ...args)
```

though now that I look at it closer, it really should take the index first -- and probably also be curried, so that the index can be partially applied.